hivemq4/k8s-image/Dockerfile

```diff
@@ -1,46 +1,41 @@
+
+
+# Additional build image to unpack the zip files and change the permissions without retaining large layers just for those operations
+FROM alpine:3.19@sha256:c5b1261d6d3e43071626931fc004f70149baeba2c8ec672bd4f27761f8e1ad6b as unpack
+
 ARG PROMETHEUS_EXTENSION_VERSION=4.0.8
 
-FROM alpine:3.19@sha256:c5b1261d6d3e43071626931fc004f70149baeba2c8ec672bd4f27761f8e1ad6b as BUILD
-ARG PROMETHEUS_EXTENSION_VERSION
+RUN mkdir -p /opt/hivemq/extensions
 
-WORKDIR /opt/hivemq/extensions
+COPY hivemq-k8s-sync-extension-*.zip /tmp/hivemq-k8s-sync-extension.zip
+RUN unzip /tmp/hivemq-k8s-sync-extension.zip -d /opt/hivemq/extensions \
+    && rm /tmp/hivemq-k8s-sync-extension.zip \
+    && chmod -R 770 /opt/hivemq/extensions/hivemq-k8s-sync-extension
 
-COPY hivemq-k8s-sync-extension-*.zip ./hivemq-k8s-sync-extension.zip
-RUN unzip hivemq-k8s-sync-extension.zip -d /opt/hivemq/extensions/ \
-    && rm hivemq-k8s-sync-extension.zip
+RUN apk --update add curl && curl -L https://github.com/hivemq/hivemq-prometheus-extension/releases/download/${PROMETHEUS_EXTENSION_VERSION}/hivemq-prometheus-extension-${PROMETHEUS_EXTENSION_VERSION}.zip -o /tmp/hivemq-prometheus-extension.zip \
+    && unzip /tmp/hivemq-prometheus-extension.zip -d /opt/hivemq/extensions \
+    && rm /tmp/hivemq-prometheus-extension.zip \
+    && rm /opt/hivemq/extensions/hivemq-prometheus-extension/prometheusConfiguration.properties
+COPY prometheusConfiguration.properties /opt/hivemq/extensions/hivemq-prometheus-extension/prometheusConfiguration.properties
+RUN chmod -R 770 /opt/hivemq/extensions/hivemq-prometheus-extension
 
-RUN apk --update add curl && curl -sLO https://github.com/hivemq/hivemq-prometheus-extension/releases/download/${PROMETHEUS_EXTENSION_VERSION}/hivemq-prometheus-extension-${PROMETHEUS_EXTENSION_VERSION}.zip \
-    && unzip hivemq-prometheus-extension-${PROMETHEUS_EXTENSION_VERSION}.zip -d /opt/hivemq/extensions/ \
-    && rm hivemq-prometheus-extension-${PROMETHEUS_EXTENSION_VERSION}.zip /opt/hivemq/extensions/hivemq-prometheus-extension/prometheusConfiguration.properties
-COPY prometheusConfiguration.properties /opt/hivemq/extensions/hivemq-prometheus-extension/
 
-## Clean image to distribute
+# Actual image
 FROM ${BASEIMAGE}
 
 ENV HIVEMQ_LOG_LEVEL=INFO\
     HIVEMQ_CLUSTER_TRANSPORT_TYPE=TCP
 
 COPY config.xml /opt/hivemq/conf/config.xml
+COPY 15_k8s_entrypoint.sh /docker-entrypoint.d/15_k8s_entrypoint.sh
 COPY scripts/*.sh /opt/hivemq/bin/
-COPY --from=BUILD /opt/hivemq/extensions/ /opt/hivemq/extensions/
+COPY --from=unpack /opt/hivemq/extensions/ /opt/hivemq/extensions/
 
-### Move the entrypoint into the correct position & adjust for whether the base image uses entrypoint.d or not
 ### Fix permissions as --chmod on COPY is not really working at the moment and requires buildkit
 RUN chmod +x /opt/hivemq/bin/*.sh \
-    && if [ -d /docker-entrypoint.d ]; then \
-      mv /opt/hivemq/bin/pre-entry_1.sh /docker-entrypoint.d/15_k8s_entrypoint.sh && \
```

Comment:
Just out of curiosity, how was this working before? How did the `pre-entry_1.sh` make it there in the first place?

Reply:
because of this: https://github.com/hivemq/hivemq4-docker-images/blob/master/hivemq4/k8s-image/Dockerfile#L25

Reply:
because of this: https://github.com/hivemq/hivemq4-docker-images/blob/master/hivemq4/k8s-image/Dockerfile#L25